### ISSUE TITLE
Coefficients for "coif5" wavelet

### ISSUE THREAD

**Issue by pierrepaleo** [PyWavelets/pywt#247]
Can it be that there is a problem with the "coif5" coefficients ?

In the nigma/pywt version, the low-pass decomposition coefficients are
```python
print(pywt.Wavelet('coif5').filter_bank[0])
[-9.517657273819165e-08, -1.6744288576823017e-07, 2.0637618513646814e-06, 3.7346551751414047e-06, -2.1315026809955787e-05, -4.134043227251251e-05, 0.00014054114970203437, 0.00030225958181306315, -0.0006381313430451114, -0.0016628637020130838, 0.0024333732126576722, 0.006764185448053083, -0.009164231162481846, -0.01976177894257264, 0.03268357426711183, 0.0412892087501817, -0.10557420870333893, -0.06203596396290357, 0.4379916261718371, 0.7742896036529562, 0.4215662066908515, -0.05204316317624377, -0.09192001055969624, 0.02816802897093635, 0.023408156785839195, -0.010131117519849788, -0.004159358781386048, 0.0021782363581090178, 0.00035858968789573785, -0.00021208083980379827]
```

In pywavelets 0.5 (and possibly previous versions):
```python
print(pywt.Wavelet('coif5').filter_bank[0])
[-9.604010112767894e-08, -1.6237995172048338e-07, 2.0612203985788783e-06, 3.7007277113394796e-06, -2.1270221672515614e-05, -4.12198619242655e-05, 0.00014035632812373243, 0.0003018579416682448, -0.0006375589261258812, -0.0016616273039298788, 0.0024315754425382886, 0.006761520220620417, -0.009159507338676163, -0.019758391600965465, 0.032674799467057355, 0.041287530472117834, -0.10556315130733723, -0.06203775157498196, 0.4379823066591634, 0.7742936228603274, 0.42157126673075435, -0.052046670253554764, -0.09192158806008609, 0.028169744270532353, 0.023408322118927783, -0.010131584846900276, -0.00415931262757864, 0.0021782943778456947, 0.0003585777411617577, -0.000212081862067494]
```

I also checked with Matlab R2013a and R2016a.
The other coefficients (including coif1-4) seem to be fine, though.

**Comment by grlee77** [comment]
I think some of the coefficient calculations were refactored for release 0.5.  I haven't had a chance to look into this yet, but it seems like they agree up to 7-8 decimal places?  Is the agreement better than this for the other coif wavelets?
@holgern or @kwohlfahrt, are you available to look into this?

**Comment by pierrepaleo** [comment]
I agree that they are quite close anyway. However it looks like slightly worse for coif5:
```python
# nigma/pywt version
import pywt
import pickle
D = {}
for wname in pywt.wavelist():
	D[wname] = pywt.Wavelet(wname).filter_bank[0]
with open("filterbanks.dict", "wb") as fid:
	pickle.dump(D, fid)

# PyWavelets 0.5
import pywt
import pickle
with open("filterbanks.dict", "rb") as fid:
	D = pickle.load(fid)
maxerr = {}
for wname in D.keys():
	maxerr[wname] = np.max(np.abs(np.array(D[wname]) - np.array(pywt.Wavelet(wname).filter_bank[0])))
```
Then, `maxerr` is less than `1E-11` for most wavelets, but it is `1.1E-05` for `coif5`.
This made one of my unit tests fail for this wavelet.

I am not expert at all regarding the coefficients computation, I am curious to know why and how the coefficients were recomputed ? Which "should" be used ?

**Comment by grlee77** [comment]
I took a brief look at this and there are definitely fewer significant digits retained in Matlab's coefficient tables.  However, it is not just a matter of truncation, but the values retained actually differ after the first several digits:
e.g. the first coefficient in Matlab is `-0.00014996380000` while the tables PyWavelets is using has `-1.499645228345950331670593167919531667975440598691604525531231e-04`

From the docstring, the values in PyWavelets are those tabulated by Kazuo Hatano at the Aichi Instiute of Technology.  These are available in PDF form at http://repository.kulib.kyoto-u.ac.jp/dspace/handle/2433/60181

Unfortunately that document is in Japanese, but I have found other references to it in English as an accurate tabulation of coefficients up to as many as 50 digits.  Assuming there has not been an error in the tabulation or transcription, the PyWavelets coefficients should be more accurate than those provided by Matlab.

I will see if I can locate another independent source to compare to.
